### PR TITLE
fix: zero trading amount

### PR DESCRIPTION
**Pull request by jsy1218**
we have some division by zero runtime error, due to trading amount being zero. trading amount cannot be zero, so we fail fast, by returning 400.

https://app.warp.dev/block/WPJHJVDpVHxYNbp6VLultL

### CODE REVIEW

**Comment by jsy1218** [comment]
* **#903** <a href="https://app.graphite.dev/github/pr/Uniswap/routing-api/903?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jsy1218 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: lib/handlers/quote/quote.ts [hunk 1]
```diff
@@ -416,6 +424,14 @@ export class QuoteHandler extends APIGLambdaHandler<
       case 'exactOut':
         amount = CurrencyAmount.fromRawAmount(currencyOut, JSBI.BigInt(amountRaw))
 
+        if (!amount.greaterThan(CurrencyAmount.fromRawAmount(currencyIn, JSBI.BigInt(0)))) {
+          return {
+            statusCode: 400,
```
Comment (xrsv) [inline lib/handlers/quote/quote.ts:429]: is `400` recognized by upstream? i see we return `404` for no liquidity

**Review by xrsv — APPROVED** [review]
🦄 - looks good, left one comment

File: lib/handlers/quote/quote.ts (hunk 1 above)
Comment (jsy1218) [reply lib/handlers/quote/quote.ts:429]: yea URA will throw non-404 errors for [retryability](https://github.com/Uniswap/backend/blob/d7ef4b2af406ece6faf4311d3ba898e2877dc855/packages/services/unified-routing-api/lib/providers/quoters/RoutingApiQuoter.ts#L117). 

we have another [quote test with amount negative](https://github.com/Uniswap/routing-api/blob/c8b8fdca4399643083c007943c421768cacf5d6e/test/mocha/e2e/quote.test.ts#L2573), that asserts 400 as well.

**Comment by jsy1218** [comment]
### Merge activity

* **Nov 4, 2:15 PM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/Uniswap/routing-api/903).
